crates/uv/src/commands/tool/run.rs

```diff
@@ -294,48 +297,84 @@ fn warn_executable_not_provided_by_package(
     }
 }
 
-/// Parse a target into a command name and a requirement.
-fn parse_target(target: &OsString) -> anyhow::Result<(Cow<OsString>, Cow<str>)> {
-    let Some(target_str) = target.to_str() else {
-        return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
-    };
+#[derive(Debug, Clone)]
+enum Target<'a> {
+    /// e.g., `ruff`
+    Unspecified(&'a str),
+    /// e.g., `ruff@0.6.0`
+    Version(&'a str, Version),
+    /// e.g., `ruff@latest`
+    Latest(&'a str),
+    /// User-defined
```

Comment:
I didn't know what this meant without reading the parser. Can you make this a bit clearer?